subprojects/file-watching/src/test/groovy/org/gradle/internal/watch/vfs/impl/WatchRootUtilTest.groovy

```diff
@@ -80,11 +81,11 @@ class WatchRootUtilTest extends Specification {
     }
 
     private static RegularFileSnapshot fileSnapshot(String absolutePath) {
-        new RegularFileSnapshot(absolutePath, absolutePath.substring(absolutePath.lastIndexOf('/') + 1), Hashing.md5().hashString(absolutePath), new FileMetadata(1, 1))
+        new RegularFileSnapshot(absolutePath, absolutePath.substring(absolutePath.lastIndexOf('/') + 1), Hashing.md5().hashString(absolutePath), new FileMetadata(1, 1), AccessType.DIRECT)
     }
 
     private static CompleteDirectorySnapshot directorySnapshot(String absolutePath) {
-        new CompleteDirectorySnapshot(absolutePath, PathUtil.getFileName(absolutePath), [], Hashing.md5().hashString(absolutePath))
+        new CompleteDirectorySnapshot(absolutePath, PathUtil.getFileName(absolutePath), [], Hashing.md5().hashString(absolutePath), AccessType.DIRECT)
     }
```

Comment:
Shall we add tests with symlinked directories for this one?

Reply:
No, since `WatchRootUtil` should never be called for snapshots accessed via symlinks.

Reply:
Why not? If the project directory is a symlink, wouldn't that be a symlinked root for example?

Reply:
It would, but we wouldn't even try to watch it since we filter out the symlinked roots before that.